### PR TITLE
Add pprof and create admin endpoint

### LINKED ISSUE #1315: No way of profiling jaeger-collector
Hi,
I was analyzing performance of jaeger-collector to see what objects are allocated and what are the method times after running jaeger under high load.

 I've noticed there is no pprof endpoints registered, so I've had to add it and build image myself.

Are there any methods (other than pprof) that you use for profiling?

I can create PR with my changes.

Something similar to this:
```
r := mux.NewRouter()
r.HandleFunc("/debug/pprof/", pprof.Index)
r.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
r.HandleFunc("/debug/pprof/profile", pprof.Profile)
r.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
r.HandleFunc("/debug/pprof/trace", pprof.Trace)
r.Handle("/debug/pprof/goroutine", pprof.Handler("goroutine"))
r.Handle("/debug/pprof/heap", pprof.Handler("heap"))
r.Handle("/debug/pprof/threadcreate", pprof.Handler("threadcreate"))
r.Handle("/debug/pprof/block", pprof.Handler("block"))

apiHandler := app.NewAPIHandler(jaegerBatchesHandler)
apiHandler.RegisterRoutes(r)
```

Regards, Konrad G

### PR DESCRIPTION
**Pull request by konradgaluszka**
Resolves #1315 

Refactored healthcheck package to have admin endpoint with both healthcheck and pprof endpoints

### CODE REVIEW

**Comment by yurishkuro** [comment]
looks good overall. Please get the build green (`make fmt`, etc.)

File: pkg/adminendpoint/adminendpoint.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
```
Comment (yurishkuro) [inline pkg/adminendpoint/adminendpoint.go]: add another line above: `Copyright (c) 2019 The Jaeger Authors.`

File: pkg/adminendpoint/adminendpoint_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
```
Comment (yurishkuro) [inline pkg/adminendpoint/adminendpoint_test.go]: add another line above: `Copyright (c) 2019 The Jaeger Authors.`

File: pkg/adminendpoint/adminendpoint.go (hunk 1 above)
Comment (jpkrohling) [reply pkg/adminendpoint/adminendpoint.go]: Shouldn't it be only "The Jaeger Authors" for new files, without "Uber Technoligies, Inc"?

Comment (yurishkuro) [reply pkg/adminendpoint/adminendpoint.go]: Yes, but a portion of this file is a copy

**Comment by yurishkuro** [comment]
Hm, I thought we had some bot that would add lint failures as comments on the appropriate lines.

The build is failing with:
```
Lint Failures
/home/travis/gopath/src/github.com/jaegertracing/jaeger/pkg/adminendpoint/adminendpoint.go:31:1: comment on exported type AdminEndpoint should be of the form "AdminEndpoint ..." (with optional leading article)
/home/travis/gopath/src/github.com/jaegertracing/jaeger/pkg/adminendpoint/healthcheck/healthcheck.go:100:1: comment on exported method HealthCheck.GetHandlerFunc should be of the form "GetHandlerFunc ..."
```

**Comment by konradgaluszka** [comment]
There are still some build errors, I will check it..

**Comment by yurishkuro** [comment]
restarted crossdock integration test 3 times, it keeps failing, might be something to do with this PR.

**Comment by yurishkuro** [comment]
e.g. the renaming of the health port CLI flag

**Comment by konradgaluszka** [comment]
The build is not stable for some reason. I've triggered it 3 times and got 3 different results. Crossdock was failing on my local machine 50% of the time even without my changes.

**Comment by pavolloffay** [comment]
@konradgaluszka did you manage to get stable builds? I have restarted travis and it seems to be fine.

@yurishkuro could this be merged?

**Comment by konradgaluszka** [comment]
> @konradgaluszka did you manage to get stable builds? I have restarted travis and it seems to be fine.

I am afraid build is still unstable. I was running the build locally on master branch (before my change) and still got failures in crossdock:
`Executing Matrix...

✗ [endtoend] test_driver→ (services=go) ⇒ Fail: could not retrieve traces from query service
✓ [endtoend] test_driver→ (services=java) ⇒ Pass
✓ [endtoend] test_driver→ (services=node) ⇒ Pass
✓ [endtoend] test_driver→ (services=python) ⇒ Pass
✓ [endtoend] test_driver→ (services=zipkin-brave-json) ⇒ Pass
✓ [endtoend] test_driver→ (services=zipkin-brave-json-v2) ⇒ Pass
✓ [endtoend] test_driver→ (services=zipkin-brave-thrift) ⇒ Pass
`
Do you have consistent runs when running  ./scripts/travis/build-crossdock.sh?

**Comment by objectiser** [comment]
This test seems to be unstable, not related to this PR:
```
--- FAIL: TestSaramaConsumerWrapper_start_Messages (0.01s)
    consumer_test.go:161: PASS:	Process(*consumer.saramaMessageWrapper)
    metricstest.go:49: 
        	Error Trace:	metricstest.go:49
        	            				metricstest.go:37
        	            				consumer_test.go:167
        	Error:      	Not equal: 
        	            	expected: int(0)
        	            	actual  : int64(1)
        	Test:       	TestSaramaConsumerWrapper_start_Messages
        	Messages:   	expected metric name: sarama-consumer.partitions-held, tags: map[]
```

File: cmd/flags/flags.go [hunk 3]
```diff
@@ -23,18 +23,19 @@ import (
...
 
 const (
-	spanStorageType     = "span-storage.type" // deprecated
-	logLevel            = "log-level"
-	configFile          = "config-file"
-	healthCheckHTTPPort = "health-check-http-port"
+	spanStorageType       = "span-storage.type" // deprecated
+	logLevel              = "log-level"
+	configFile            = "config-file"
+	adminEndpointHTTPPort = "admin-endpoint-http-port"
```
Comment (jpkrohling) [inline cmd/flags/flags.go]: This seem to be a breaking change. Would it be possible to make it backwards compatible? There's a `RegisterAlias()` function in `viper` that could be used for this purpose: https://github.com/spf13/viper#registering-and-using-aliases

Comment (objectiser) [reply cmd/flags/flags.go]: May be best to just handle in the same way as other deprecated flags? e.g. https://github.com/jaegertracing/jaeger/blob/master/cmd/agent/app/reporter/tchannel/flags.go#L54-L66 - so accept the flag, log warning and copy over the value.

**Comment by konradgaluszka** [comment]
> LGTM, just need to either document the breaking change or provide a backwards compatible option.

I couldn't use aliasing in viper, because it does not work with reading from config:
https://github.com/spf13/viper/issues/560

**Comment by jpkrohling** [comment]
@konradgaluszka, looks like this last commit failed the DCO. Would you please amend it?

**Comment by yurishkuro** [comment]
Thanks!

**Pull request merged**